main/src/ca/uwaterloo/flix/language/ast/RigidityEnv.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 Matthew Lutze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.ast
+
+import scala.collection.immutable.SortedSet
+
+/**
+  * Environment tracking the rigidity of type variables.
```

Comment:
Describe the invariant, i.e. only rigid variables are in the set. Others are flexible.